src/main.rs

```diff
@@ -237,10 +239,13 @@ impl<WATCH: Watch, WRITE: Write> Monitor<WATCH, WRITE> {
 
                     for (id, replica) in self.replicas.iter_mut() {
                         for path in &paths {
-                            for dir in &replica.paths {
-                                if let Ok(relative_path) = path.strip_prefix(dir) {
+                            for (dir, subdir_name) in &replica.paths {
+                                if let Ok(relative_path) = path.strip_prefix(&dir) {
                                     matched_replica_ids.insert(id);
-                                    replica.pending_changes.insert(relative_path.into());
+                                    replica.pending_changes.insert(match subdir_name {
+                                        None    => relative_path.into(),
+                                        Some(s) => s.join(relative_path).into(),
```

Comment:
This won't work as `START` command is also issued whenever there is a new dir created. This captured subdir would only be the last subdir. It is not guaranteed this subdir should be applied to every change in the replica.

Can you actually provide reproduction steps for the issue you're trying to solve?

Reply:
Following is the example configuration where current `unison-fsmonitor` fails:

    root = /Users/user/projects/
    root = ssh://host.tld//home/user/
    path = workspace
    ignore = Name *.tmp
    ignore = Name .git
    perms = 0
    dontchmod = false
    batch = true

Intent of above configuration is to only synchronize `workspace` subdirectory. A workaround is stop using `path`, and instead append `workspace` to `root` values.

When `unison-fsmonitor` detects changes in `workspace/foo` file it reports them as `foo` which `unison` ignores because I believe there is no way for unison to figure out `path` this `foo` corresponds to (for there could be many `path`-s). With the diff in this PR, `unison-fsmonitor` prepends the `path` which `unison` happily accepts and synchronizes.

HTH

Reply:
Thanks. I got this reproduced locally.

Reply:
I'm still have a concern regarding the fix though.

As I described earlier, when `path` is not specified, unison would send `START id root dir` whenever a new dir is created to append the dir to watched list. The change here will break in this case.

I will submit a PR. Please confirm it solves your issue.

Reply:
Please check https://github.com/autozimu/unison-fsmonitor/pull/4.

Reply:
With following confguration, and my commit:

```root = /Users/user/projects/testing/
root = ssh://host.tld//home/user/testing/
path = foo
path = bar
perms = 0
dontchmod = false
batch = true
```

I tried creating/updating/remove permutations of subdirectories/files of both `foo`, and `bar` subdirectories on both remote and local side, and in all the cases changes seem to get propagated seemlessly.

I even tried with above configuration (minus `path` lines), and same changes and that also seem to work without any issues.

Looking at the source code, it seems like when `unison` sends a `START` for a newly created sub-directory we skip tracking it because we are already (recursively) monitoring the parent directory, which is why what I did above seem to work fine.

So it seems to me, I don't understand the problem you are describing. If it's not too much inconvenience on your part, maybe share some precise steps to reproduce.

Reply:
You're correct.

I myself overlooked the checking of `contains_path`.